drivers/SmartThings/matter-lock/src/new-matter-lock/init.lua

```diff
@@ -0,0 +1,1615 @@
+  end
+  for _, p in ipairs(NEW_MATTER_LOCK_PRODUCTS) do
+    if device.manufacturer_info.vendor_id == p[1] and
+      device.manufacturer_info.product_id == p[2] then
+        return true
+    end
+  end
+  return false
+end
+
+local function find_default_endpoint(device, cluster)
+  local res = device.MATTER_DEFAULT_ENDPOINT
+  local eps = device:get_endpoints(cluster)
+  table.sort(eps)
+  for _, v in ipairs(eps) do
+    if v ~= 0 then --0 is the matter RootNode endpoint
+      return v
+    end
+  end
+  device.log.warn(string.format("Did not find default endpoint, will use endpoint %d instead", device.MATTER_DEFAULT_ENDPOINT))
+  return res
+end
+
+local function component_to_endpoint(device, component_name)
+  return find_default_endpoint(device, clusters.DoorLock.ID)
+end
+
+local function device_init(driver, device)
+  device:set_component_to_endpoint_fn(component_to_endpoint)
+  device:subscribe()
+ end
+
+local function device_added(driver, device)
+  device:emit_event(capabilities.lockAlarm.alarm.clear({state_change = true}))
+end
+
+local function do_configure(driver, device)
+  local user_eps = device:get_endpoints(DoorLock.ID, {feature_bitmap = DoorLock.types.Feature.USER})
+  local pin_eps = device:get_endpoints(DoorLock.ID, {feature_bitmap = DoorLock.types.Feature.PIN_CREDENTIAL})
+  local week_schedule_eps = device:get_endpoints(DoorLock.ID, {feature_bitmap = DoorLock.types.Feature.WEEK_DAY_ACCESS_SCHEDULES})
+  local year_schedule_eps = device:get_endpoints(DoorLock.ID, {feature_bitmap = DoorLock.types.Feature.YEAR_DAY_ACCESS_SCHEDULES})
+
+  local profile_name = "lock"
+  if #user_eps > 0 then
+    profile_name = profile_name .. "-user"
+    if #pin_eps > 0 then
+      profile_name = profile_name .. "-pin"
+    end
+    if #week_schedule_eps + #year_schedule_eps > 0 then
+      profile_name = profile_name .. "-schedule"
+    end
+  else
+    profile_name = "base-lock"
+  end
+  device.log.info(string.format("Updating device profile to %s.", profile_name))
+  device:try_update_metadata({profile = profile_name})
+end
+
+local function info_changed(driver, device, event, args)
+  for cap_id, attributes in pairs(subscribed_attributes) do
+    if device:supports_capability_by_id(cap_id) then
+      for _, attr in ipairs(attributes) do
+        device:add_subscribed_attribute(attr)
+      end
+    end
+  end
+  for cap_id, events in pairs(subscribed_events) do
+    if device:supports_capability_by_id(cap_id) then
+      for _, e in ipairs(events) do
+        device:add_subscribed_event(e)
+      end
+    end
+  end
+  device:subscribe()
+end
+
+-- Matter Handler
+----------------
+-- Lock State --
+----------------
+local function lock_state_handler(driver, device, ib, response)
+  local LockState = DoorLock.attributes.LockState
+  local attr = capabilities.lock.lock
+  local LOCK_STATE = {
+    [LockState.NOT_FULLY_LOCKED] = attr.not_fully_locked(),
+    [LockState.LOCKED] = attr.locked({visibility = {displayed = false}}),
```

Comment:
Why is `displayed = false` here?

Reply:
the change of lock state occurs twice from lock_state_handler and lock_op_event_handler. so I wanted to add "displayed is false". but now I updated it to add call_with_delay to prevent duplicated event occurs.